src/core/lib/event_engine/posix_engine/posix_engine.h

```diff
@@ -138,6 +139,9 @@ class PosixEventEngine final : public PosixEventEngineWithFdSupport,
  public:
   class PosixDNSResolver : public EventEngine::DNSResolver {
    public:
+    explicit PosixDNSResolver(ResolverOptions const& options,
+                              PosixEnginePollerManager* poller_manager,
+                              PosixEventEngine* event_engine);
```

Comment:
Does the resolver need to keep the engine alive? Chances are that all resolutions will be completed by the time an EventEngine wants to be destroyed, but we may have flakes for edge cases.

Reply:
My concern is that we might not have the test cases to cover a possible real-world ordering of lifetime extinctions, but I'm not certain. It's worth thinking about a bit more. If the resolver owned a `shared_ptr<PosixEventEngine>`, then we wouldn't have to think about it.